tests/test_diffraction_objects.py

```diff
@@ -212,21 +213,41 @@ def _test_valid_diffraction_objects(actual_diffraction_object, function, expecte
 
 
 def test_get_angle_index():
-    test = DiffractionObject()
-    test.angles = np.array([10, 20, 30, 40, 50, 60])
-    actual_angle_index = test.get_angle_index(angle=10)
-    assert actual_angle_index == 0
+    test = DiffractionObject(
+        wavelength=0.71, xarray=np.array([30, 60, 90]), yarray=np.array([1, 2, 3]), xtype="tth"
+    )
+    actual_index = test.get_array_index(xtype="tth", value=30)
```

Comment:
we need more cases.  Here everything is integers and there is a match.  What do we want to happen if the value lies between two other values?  Return nearest and a warning?  What if it lies outside the range of values?  Return nearest and a warning?  What if it is really far away?  Have a threshold after which we raise and error?

Reply:
@sbillinge I addressed this in the new PR. please review.